flexmeasures/data/services/scheduling.py

```diff
@@ -241,6 +246,92 @@ def create_scheduling_job(
     return job
 
 
+def cb_done_sequential_scheduling_job(jobs_ids: list[str]):
+    """
+    TODO: add logic
+    """
+
+    # jobs = [Job.fetch(job_id) for job_id in jobs_ids]
+    pass
+
+
+def create_sequential_scheduling_job(
+    asset_or_sensor: Asset | Sensor | None = None,
+    job_id: str | None = None,
+    enqueue: bool = True,
+    requeue: bool = False,
+    force_new_job_creation: bool = False,
+    scheduler_specs: dict | None = None,
+    depends_on: list[Job] | None = None,
+    **scheduler_kwargs,
+):
+    flex_model = scheduler_kwargs["flex_model"]
+    jobs = []
+    previous_sensors = []
+    previous_job = depends_on
+    for child_flex_model in flex_model:
+        sensor = child_flex_model.pop("sensor")
+
+        current_scheduler_kwargs = deepcopy(scheduler_kwargs)
+
+        current_scheduler_kwargs["flex_model"] = child_flex_model["sensor_flex_model"]
+        if "inflexible-device-sensors" not in current_scheduler_kwargs["flex_context"]:
+            current_scheduler_kwargs["flex_context"]["inflexible-device-sensors"] = []
+        current_scheduler_kwargs["flex_context"]["inflexible-device-sensors"].extend(
+            previous_sensors
+        )
+        current_scheduler_kwargs["resolution"] = sensor.event_resolution
+        current_scheduler_kwargs["sensor"] = sensor
+
+        job = create_scheduling_job(
+            **current_scheduler_kwargs,
+            scheduler_specs=scheduler_specs,
+            requeue=requeue,
+            job_id=job_id,
+            enqueue=False,  # we enqueue all jobs later in this method
```

Comment:
I'm not too sure about my change here, but I believe I saw duplicate jobs in the queue otherwise.

Reply:
We are enqueuing them later altogether.  We are not checking the function argument, though. This is something to be fixed.